### PR TITLE
Select Positions not working sometimes

### LINKED ISSUE #198: [FE] Can't load positions the first time
In the select modal positions, the first time is like the positions can't be loaded. But if you try again, works. The feature must work the first time...
Peek 07-09-2020 20-21

### LINKED ISSUE #239: 'Custom tokens' icons are not shown in the Positions List
Related to #238 , #175

See the screenshot: 

Also, 'Custom tokens' icons are not shown in the **Select Positions pop-ups**

### LINKED ISSUE #244: [BUG / FE] Split Position: Select Position
A nasty bug when selecting a position for the "Split From" field on the "Split Position" section.

Found it on https://pr237--conditionaltokensexplorer.review.gnosisdev.com/#/split but `/develop` is the same.

Sep-14-2020 14-08-40

### PR DESCRIPTION
**Pull request by hpmaxi**
Closes #198 
Closes #244 
Closes #239 

This fixes a race condition but also tries to make custom tokens a working feature on each page that could use them, trying to hide the details of wrangling positions data inside a hook and moving common logic to utils (duplicated in `usePositions` and `useCollateral`).

This feature could be improved caching custom tokens already fetched.

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr236--conditionaltokensexplorer.review.gnosisdev.com

**Comment by elena-zh** [comment]
I have reviewed the changes. 
Currently, when a user calls Positions modal to select any for the 1st time, the loader is shown. Then it is closed, empty modal is displayed for 1-3 sec. The the list of positions appear. When I call the positions pop-up once again, empty grid 'blinks' before the whole list is loaded. 
https://drive.google.com/file/d/1dM5l8gl75YRek4x0Da-n3iO-lPu51ZVb/view

Can we show a loader until the while positions list is loaded (and not to show an empty pop-up)?

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr236--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr236--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr236--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr236--conditionaltokensexplorer.review.gnosisdev.com

File: src/components/mergePositions/MergePreview/index.tsx [hunk 1]
```diff
@@ -10,29 +10,38 @@ import { TitleValue } from 'components/text/TitleValue'
...
     if (canMergePositions && condition && positions.length > 0) {
-      const token = networkConfig.getTokenFromAddress(positions[0].collateralToken.id)
-      return getMergePreview(positions, condition, amount, token)
+      getTokenSummary(networkConfig, provider, positions[0].collateralToken.id).then((token) => {
+        if (!cancelled) {
+          setMergedPosition(getMergePreview(positions, condition, amount, token))
+        }
+      })
+    } else {
+      setMergedPosition('')
```
Comment (hpmaxi) [inline src/components/mergePositions/MergePreview/index.tsx:39]: Maybe this could also use `useWithToken`

**Review by elena-zh — APPROVED** [review]

**Review by gabitoesmiapodo — COMMENTED** [review]
It works.

However, it would be good to have an extra set of eyes looking at the code, so I'll let @mariano-aguero have the final say about this.

**Comment by mariano-aguero** [comment]
@hpmaxi @gabitoesmiapodo I found something really weird, look at the image... is like the first icon is displayed, but the rest no..
Screenshot_20200915_135839

**Comment by mariano-aguero** [comment]
Not related with this issue, If we select a position I think should say `Use position balance` instead of `Use wallet balance`

Screenshot_20200915_140151

Thoughts @elena-zh @gabitoesmiapodo ?

File: src/hooks/useWithToken.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        })
+      )
+    }
+
+    if (data && data.length > 0) {
+      setLoading(true)
+      fetchTokens(data).then((withToken) => {
+        if (!cancelled) {
+          setDataWithToken(withToken)
+          setLoading(false)
```
Comment (mariano-aguero) [inline src/hooks/useWithToken.tsx:41]: Mmm maybe is possible that Loading will be always true, is just a concern

**Review by mariano-aguero — COMMENTED** [review]
I left some comments

File: src/hooks/useWithToken.tsx (hunk 2 above)
Comment (hpmaxi) [reply src/hooks/useWithToken.tsx:41]: Maybe a catch or finally that check if its cancelled and set loading to false. What do you think @mariano-aguero ?

**Comment by mariano-aguero** [comment]
@hpmaxi please see the comments and solve the conflicts

**Comment by mariano-aguero** [comment]
> @hpmaxi @gabitoesmiapodo I found something really weird, look at the image... is like the first icon is displayed, but the rest no..
> Screenshot_20200915_135839

@hpmaxi If you can't reproduce this one, please create an issue so we can work this more isolate

**Review by mariano-aguero — APPROVED** [review]

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr236--conditionaltokensexplorer.review.gnosisdev.com

**Pull request merged**